pkg/crd/manifest.go

```diff
@@ -14,28 +14,63 @@
  * limitations under the License.
  */
 
-package core
+package crd
 
 import (
 	"errors"
 	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/jinzhu/copier"
+	"github.com/projectriff/riff/pkg/fileutils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"net/url"
 	"path/filepath"
+	"strings"
+)
 
-	"github.com/projectriff/riff/pkg/fileutils"
+type ResourceChecks struct {
+	Kind     string               `json:"kind,omitempty"`
+	Selector metav1.LabelSelector `json:"selector,omitempty"`
+	JsonPath string               `json:"jsonpath,omitempty"`
+	Pattern  string               `json:"pattern,omitempty"`
+}
 
-	"github.com/ghodss/yaml"
-)
+type Resource struct {
+	Path      string           `json:"path,omitempty"`
+	Content   string           `json:"content,omitempty"`
+	Name      string           `json:"name,omitempty"`
+	Namespace string           `json:"namespace,omitempty"`
+	Checks    []ResourceChecks `json:"checks,omitempty"`
+}
 
-const manifestVersion_0_1 = "0.1"
+type RiffResource struct {
+	System         []Resource `json:"system,omitempty"`
+	Initialization []Resource `json:"initialization,omitempty"`
+}
+
+type RiffSpec struct {
+	Images    []string     `json:"images,omitempty"`
+	Resources RiffResource `json:"resources,omitempty"`
+}
+
+type RiffStatus struct {
+	Status string `json:"status,omitempty"`
+}
 
-// Manifest defines the location of YAML files for system components.
 type Manifest struct {
-	ManifestVersion string   `json:"manifestVersion"`
-	Istio           []string `json:"istio"`
-	Knative         []string `json:"knative"`
-	Namespace       []string `json:"namespace"`
-	manifestDir     string
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec        RiffSpec   `json:"spec,omitempty"`
+	Status      RiffStatus `json:"status,omitempty"`
+	manifestDir string
+}
+
+func (orig *Manifest) DeepCopyObject() runtime.Object {
+	result := &Manifest{}
+	copier.Copy(result, orig)
```

Comment:
Is it worth introducing a new dependency for a deep copy?

k8s.io/code-generator generated clients will also generate the DeepCopy method.